# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/BuildConstants.h
# Commit: 05896dbeb434
# Full Hash: 05896dbeb434d6f3ca9324f01eb67020d932f50f
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Determine webgl's layers::TextureType in Present. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79617
# ==============================================================================

diff -r 8439f89e170b -r 05896dbeb434 gfx/layers/BuildConstants.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/layers/BuildConstants.h	Mon Jun 15 18:26:14 2020 +0000
@@ -0,0 +1,57 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BUILD_CONSTANTS_H_
+#define BUILD_CONSTANTS_H_
+
+/**
+ * Why not just use ifdefs?
+ * ifdefs tend to result in code that compiles on one platform but not another.
+ * Given the number of build and platform configurations we have, it's best to
+ * aim to compile the same on as many platforms as possible, and let the
+ * compiler see the constexprs and handle dead-code elision itself.
+ */
+
+namespace mozilla {
+
+constexpr bool kIsWindows =
+#ifdef XP_WIN
+    true;
+#else
+    false;
+#endif
+
+constexpr bool kIsMacOS =
+#ifdef XP_MACOSX
+    true;
+#else
+    false;
+#endif
+
+constexpr bool kIsX11 =
+#ifdef MOZ_X11
+    true;
+#else
+    false;
+#endif
+
+constexpr bool kIsWayland =
+#ifdef MOZ_WAYLAND
+    true;
+#else
+    false;
+#endif
+
+constexpr bool kIsAndroid =
+#ifdef MOZ_WIDGET_ANDROID
+    true;
+#else
+    false;
+#endif
+
+}  // namespace mozilla
+
+#endif  // BUILD_CONSTANTS_H_
\ No newline at end of file